src/Microsoft.DncEng.SecretManager/StorageUtils.cs

```diff
@@ -74,22 +75,54 @@ public static SharedAccessAccountPermissions AccountPermissionsFromString(string
         public static string GenerateBlobAccountSas(string connectionString, string permissions, string service, DateTimeOffset expiryTime)
         {
             CloudStorageAccount account = CloudStorageAccount.Parse(connectionString);
+            SharedAccessAccountServices serviceList = default(SharedAccessAccountServices);
+            if(service.Contains("|"))
+            {
+                HashSet<string> servicesUsed = new HashSet<string>();
+                foreach(var serviceString in service.Split("|"))
+                {
+                    if(!servicesUsed.Add(serviceString))
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(service));
+                    }
+                    switch(serviceString)
+                    {
+                        case "blob": serviceList |= SharedAccessAccountServices.Blob;
+                            break;
+                        case "table": serviceList |= SharedAccessAccountServices.Table;
+                            break;
+                        case "file": serviceList |= SharedAccessAccountServices.File;
+                            break;
+                        case "queue": serviceList |= SharedAccessAccountServices.Queue;
+                            break;
+                        default: throw new ArgumentOutOfRangeException(nameof(service));
+                    }
+                }
+            }
+            else
+            {
+                switch(service)
```

Comment:
```suggestion
                switch(service.ToLowerInvariant())
```
To match old behavior.